Remove direct use of Object Manager

AnshuMishra17 opened the pull request:
Description
Remove direct use of Object Manager by adding dependency in constructor.

Manual testing scenarios
Go to Content -> CMS Pages
Delete CMS pages from edit page and massaction as well.

AnshuMishra17 committed:
Remove direct use of Object Manager

Remove direct use of Object Manager by adding dependency in constructor.

AnshuMishra17 committed:
Merge branch '1.2-develop' of https://github.com/AnshuMishra17/magento2 into 2.2-develop

ihor-sviziev commented on app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php:
```diff
@@ -30,8 +49,7 @@ public function execute()
                 // init model and delete
-                $model = $this->_objectManager->create(\Magento\Cms\Model\Page::class);
-                $model->load($id);
+                $model = $this->cmsPage->load($id);
```
1. I think in this case it's better to use repository to retrieve and then delete this cms page
2. Loading model is not correct way. It's better to use \Magento\Cms\Model\PageFactory and create one instance of model your your needs (if really needed)

AnshuMishra17 replied:
Done

AnshuMishra17 committed:
Refactor Code

Use repository to retrieve and then delete this cms page

AnshuMishra17 commented:
I have made the changes @ihor-sviziev can you please review?

ihor-sviziev commented on app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php:
```diff
@@ -14,6 +16,23 @@ class Delete extends \Magento\Backend\App\Action
+        \Magento\Cms\Api\PageRepositoryInterface $cmsPage
+    ) {
+        parent::__construct($context);
+        $this->cmsPage = $cmsPage ?: \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\Cms\Api\PageRepositoryInterface::class);
```
It's not "cms page", now it's "cms page repository", please rename it

AnshuMishra17 replied:
Done

ihor-sviziev commented on app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php:
```diff
@@ -14,6 +16,23 @@ class Delete extends \Magento\Backend\App\Action
+     */
+    public function __construct(
+        Context $context,
+        \Magento\Cms\Api\PageRepositoryInterface $cmsPage
```
Please make new dependency optional (you already added fallback to object manager) due to backward compatibility reasons

AnshuMishra17 replied:
Done.

ihor-sviziev commented:
@AnshuMishra17
1. please fix these changes and combine all your changes into one commit
2. unit tests are failing right now, please check them

AnshuMishra17 committed:
refactor code

1. Renamed "cms page" to "cms page repository".
2. Made cms page repository dependency optional

AnshuMishra17 committed:
refactor cms page delete controller

AnshuMishra17 committed:
refactor cms page delete controller

AnshuMishra17 commented:
@ihor-sviziev I have made the necessary modifications. It would be great if you can verify and let me know if there are still any issues.
I am new here, I will be thankful to you as well if you can guide me towards the correct approach and steps.
Also, how can I check what are the issues with unit tests?

AnshuMishra17 committed:
Refactor cms page code

AnshuMishra17 commented:
@ihor-sviziev How can track the issue here?
I have checked one report and it is saying "Error: Call to a member function getTitle() on null".
But I have checked using my local setup manual testing by printing the title and the deleted page title is getting print.
How can I make sure in which scenario the test is getting failed?

Jakhotiya commented:
Hi @AnshuMishra17 you will need to modify this unit test "\Magento\Cms\Test\Unit\Controller\Adminhtml\Page\DeleteTest". You have refactored the Main Controller class but you will also need to cover all those changes and update this test class.

AnshuMishra17 committed:
Update DeleteTest.php

AnshuMishra17 commented:
@ihor-sviziev @Jakhotiya How to modify Magento\Cms\Test\Unit\Controller\Adminhtml\Page\DeleteTest ? If anyone of you can guide me that will be really helpful to me.

AnshuMishra17 committed:
remove load action

AnshuMishra17 committed:
Update DeleteTest.php

AnshuMishra17 committed:
Update DeleteTest.php

AnshuMishra17 committed:
Test class compatibility

AnshuMishra17 committed:
Update DeleteTest.php

Jakhotiya commented:
@AnshuMishra17 have you run this unit test on your local? Seems like tests are failing on CI. Can you run all unit tests on local? If you are not familiar with how to do that, [unit testing in magento 2](https://devdocs.magento.com/guides/v2.2/test/unit/unit_test_execution_cli.html) should help.
Sorry for the delay in response.

AnshuMishra17 committed:
Merge branch '2.2-develop' of https://github.com/magento/magento2 into 2.2-develop

AnshuMishra17 committed:
Merge branch '2.2-develop' of https://github.com/AnshuMishra17/magento2 into 2.2-develop

AnshuMishra17 committed:
Merge branch '2.2-develop' of https://github.com/magento/magento2 into 2.2-develop

AnshuMishra17 committed:
Corrected spelling mistake in comments

Change requere to require in comments for public function addTaxPercents()
